ocaml/lambda/matching.ml

```diff
@@ -2854,12 +2855,12 @@ let combine_constructor loc arg pat_env cstr partial ctx def
                   | None, _ -> same_actions nonconsts
                 in
                 match act0 with
-                | Some act ->
+                | Some act when not (Clflags.is_flambda2 ()) ->
```

Comment:
What objection does Flambda2 have here? `Lifthenelse (Lprim (Pisint, ...), ...)` seems fine.

Reply:
I think this is a mistake actually, this code should indeed be fine.